release notes/v0.40.0.md

```diff
@@ -0,0 +1,280 @@
+- [#2582](https://github.com/grafana/k6/pull/2582) Previously, if `RegisterCallback` result was called twice, the second call would silently break the event loop. This has never been expected behavior, and calling it twice is always a bug in the code using it. Now, calling the `RegisterCallback` result twice leads to a panic.
+- [#2596](https://github.com/grafana/k6/pull/2596) The `tainted` property of the Metric type is no longer outputted by the JSON output. That property was likely always going to have a `false` value as it was outputted at the beginning of the test.
+
+## Main module/script no longer taints the global scope [#2571](https://github.com/grafana/k6/pull/2571)
+
+During the ESM changes, we found that anything defined in the main module scope was also accessible globally. This was because it was directly evaluated in the global scope.
+This has now been remedied and is no longer the case. This is a *breaking* change, but given that the whole point of modules (CommonJS or ESM) is to separate them, this is obviously rather a bug than a feature.
+
+On that note, we've seen [reports](https://github.com/grafana/k6/issues/2623) by people who have this global accessibility of the main module (intentionally or not). Still, it seems relatively rare, with only a few usages in a script. So if you need to access something globally, our suggested workaround is to set it explicitly on the global object `globalThis`.
+
+## `k6/ws` now considers the `throw` option [#2247](https://github.com/grafana/k6/issues/2247)
+
+`k6/http` has used the [`throw` option](https://k6.io/docs/using-k6/k6-options/reference/#throw) to figure out whether it should throw an exception on errors or return a response object with an error set on it (and log it).
+
+This functionality is finally also available for `k6/ws`, which previously would've *always* thrown an exception and thus involved more scripting to handle it.  [#2616](https://github.com/grafana/k6/pull/2616)
+
+This is a minor breaking change. By default, `throw` is `false`, so it now no longer throws an exception but instead returns a Response with `error` property.
+
+Thank you, @fatelei, for making this change !
+
+# New Features
+
+## Experimental modules [#2630](https://github.com/grafana/k6/pull/2630) and [#2656](https://github.com/grafana/k6/pull/2656)
+
+As mentioned in the v0.39.0 release notes, we're happy to announce that this release brings experimental modules. The main idea behind this initiative is to get community feedback earlier, which will help us improve them faster. We encourage you to try experimental modules out and provide feedback through the community forums or GitHub issues. 
+
+This release contains three experimental modules:
```

Comment:
:thinking: 

```suggestion
This release contains three experimental modules containing:
```

Reply:
Do the experimental modules contain these functionalities (modules containing...) or are the modules identical with the functionality (this release contains the modules)?

Maybe we can avoid this grammatical trickiness with:

> This release brings the following experimental modules:

Reply:
My concern is that after a sentence like `This release brings the following experimental modules:` I expected to find the names of the modules but got the list of the functionality that these modules have.

```
This release brings the following experimental modules:
- Support for interaction with [Redis](https://redis.io/)
- A new Websockets API that copies the "web" [WebSocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket)
- [setTimeout](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout) and friends.
```

Is it grammatically correct? :thinking:

Reply:
I would argue adding `containing` does not add anything. Not certain about whether it is "grammatically correct" (not exactly the right person to answer this), but I woudl argue it is still a list of the three modules it just explains what they do - arguably the thing people care about, instead of just a list of modules. 

I guess we can have them as "`k6/experimental/something` which adds ...", but I am not certain how much values that adds :shrug:

Reply:
as in https://github.com/grafana/k6/pull/2669/files#r964589089

Reply:
I agree about the value, but then probably we don't need the modules at all :thinking: 

Like:
```suggestion
This release brings the following experimental features:
```

Reply:
I do think that with the new listing of the module import path we should keep `modules` in the message. 

I guess we can still use `brings` and not say `three` :shrug:

Reply:
I mean, now we have the module names listed, so I'm good even with the current option, but if you prefer to adjust, feel free to do it :+1: :relaxed: